### ISSUE TITLE
UWP Frame in title view cause exception

### ISSUE THREAD

**Issue by alexanderzuban** [xamarin/Xamarin.Forms#9085]
### Description
If page title view has Frame switch to other page and back cause exception

### Steps to Reproduce

1. create page with custom Title View which has a Frame
2. push one more page to stack
3. pop page from stack

### Expected Behavior

original page appears

### Actual Behavior

Exception is thrown
"Value does not fall within the expected range."

### Basic Information

- Version with issue: XamarinForms 4.4

- Platform Target Frameworks: 
  - UWP:  17144

### Reproduction Link

[FrameTitleTest.zip](https://github.com/xamarin/Xamarin.Forms/files/4021341/FrameTitleTest.zip)

**Comment by WayaFlyfeather** [comment]
I think I might have run into the cause of this in a fix to #8602. Will check out, and make an isolated PR if I'm correct.

**Comment by WayaFlyfeather** [comment]
> I think I might have run into the cause of this in a fix to #8602. Will check out, and make an isolated PR if I'm correct.

No, it wasn't what I thought it might have been. It seems the bug is in the UWP FrameRenderer (it throws [from this line](https://github.com/xamarin/Xamarin.Forms/blob/6fab9871017bd08cfecfa426487cd877475b7ecb/Xamarin.Forms.Platform.UAP/FrameRenderer.cs#L77)).

Some [searching suggests](https://forums.xamarin.com/discussion/125971/system-argumentexception-crash-in-custom-uwp-framerenderer) this may not be a new bug - it may be going back to at least version 2.5(!). If it is the same it is probably not related to TitleView at all.
